src/main/java/dl/news/portal/domain/dto/UserDto.java

```diff
@@ -2,24 +2,54 @@
 
 import dl.news.portal.domain.entity.User;
 
+import javax.validation.constraints.Email;
+import javax.validation.constraints.Size;
 import java.util.Optional;
 
-public class UserDto implements DtoTransfer<User> {
-    private Optional<String> optionalUsername;
-    private Optional<String> optionalEmail;
+public class UserDto {
+    private Optional<@Size(min = 3, max = 50) String> username;
+    private Optional<
+            @Email(regexp = "[A-Za-z0-9._%-+]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,4}")
+            @Size(max = 100)
+                    String> email;
 
-    public UserDto(String username, String email) {
-        this.optionalUsername = Optional.ofNullable(username);
-        this.optionalEmail = Optional.ofNullable(email);
+    private Optional<@Size(min = 6, max = 255) String> password;
+
+    public UserDto(String username, String email, String password) {
+        this.username = Optional.ofNullable(username);
+        this.email = Optional.ofNullable(email);
+        this.password = Optional.ofNullable(password);
     }
 
     public UserDto() {
-        this(null, null);
+        this(null, null, null);
```

Comment:
absolutely  redundant constructor. Remove it

Reply:
I see